sdk/core/azure-core/src/main/java/com/azure/core/client/traits/ConnectionStringTrait.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.client.traits;
+
+/**
+ * The interface for client builders that support a connection string.
+ *
+ * @param <TBuilder> the type of client builder.
+ */
+public interface ConnectionStringTrait<TBuilder extends ConnectionStringTrait<TBuilder>> {
```

Comment:
we might also need endpoint trait (but it can be done later)

Reply:
we're less strict with 'endpoint', because we do allow other names if they are more appropriate. Of course, by and large we have done endpoint, so notifying users where endpoint is supported won't hurt.